src/extensions/emulator/specHelper.ts

```diff
@@ -94,27 +93,34 @@ export function getFunctionProperties(resources: Resource[]) {
   return resources.map((r) => r.properties);
 }
 
-export function getNodeVersion(resources: Resource[]): number {
+const DEFAULT_RUNTIME = "nodejs14";
+
+/**
+ * Get runtime associated with the resources. If conflicting, arbitrarily pick one.
+ */
+export function getRuntime(resources: Resource[]): string {
+  if (resources.length === 0) {
+    return DEFAULT_RUNTIME;
+  }
+
   const invalidRuntimes: string[] = [];
-  const versions = resources.map((r: Resource) => {
-    if (getResourceRuntime(r)) {
-      const runtimeName = getResourceRuntime(r) as string;
-      const runtime = parseRuntimeVersion(runtimeName);
-      if (!runtime) {
-        invalidRuntimes.push(runtimeName);
-      } else {
-        return runtime;
-      }
+  const runtimes = resources.map((r: Resource) => {
+    const runtime = getResourceRuntime(r);
+    if (!runtime) {
+      return DEFAULT_RUNTIME;
     }
-    return 14;
+    if (!/(nodejs)?([0-9]+)/.test(runtime)) {
+      invalidRuntimes.push(runtime);
+      return DEFAULT_RUNTIME;
+    }
+    return runtime;
   });
-
   if (invalidRuntimes.length) {
     throw new FirebaseError(
       `The following runtimes are not supported by the Emulator Suite: ${invalidRuntimes.join(
         ", "
       )}. \n Only Node runtimes are supported.`
     );
   }
-  return Math.max(...versions);
+  return runtimes[0];
```

Comment:
Nit: I'd prefer to keep this behavior of returning the highest version of whatever type of runtime we are looping through, in case we ever do switch to using the right version of node for each backend.